analyses/malawi/dryspells_trigger/data-viz.R

```diff
@@ -0,0 +1,70 @@
+
+#####
+## setup
+#####
+
+# load libraries
+library(tidyverse)
+library(sf)
+library(raster)
+
+# set options
+rasterOptions(maxmemory = 1e+09)
+options(scipen = 999)
+
+# set directory paths
+# AA_DATA_DIR is set as a variable in .Renviron or .bashprofile
+data_dir <- Sys.getenv("AA_DATA_DIR")
+shapefile_path <- paste0(data_dir, "/public/raw/mwi/cod_ab/mwi_adm_nso_20181016_shp")
+plot_path <- paste0(data_dir, "/public/processed/mwi/plots/dry_spells/dry_spell_plots")
+
+# import dry spells groundtruth dataset
+dry_spells_groundtruth_dataset_url <- "https://data.humdata.org/dataset/df68065d-704c-4556-b111-394df000cee4/resource/951c0454-23ab-4057-a7fb-7cfeed4f842d/download/full_list_dry_spells.csv"
+ds <- read.csv(dry_spells_groundtruth_dataset_url)
+
+# read in country shapefiles
+mwi_adm2 <- st_read(paste0(shapefile_path, "/mwi_admbnda_adm2_nso_20181016.shp"))
+
+#####
+## map  historical dry spells
+#####
+
+# summarise dry spells frequency per district / year
+ds_counts_per_year <- ds %>%
+  filter(season_approx != "outside rainy season") %>%
+  group_by(pcode, season_approx) %>%
+  summarise(n_ds = n())
+
+ds_counts <- ds %>%
+  filter(season_approx != "outside rainy season") %>%
```

Comment:
From the map later it seems to me that the count of number of dry spells is really high. I checked in the `01_chirps_dry_spell_detection.R` script how `outside rainy season` is defined, and it is defined as `data_long$season_approx <- ifelse(data_long$month >= 10, data_long$year, ifelse(data_long$month <= 7, data_long$year - 1, 'outside rainy season'))` 

So that is a very loose definition. We should probably change this in the HDX file.. 

I am alright with merging as it is if you are not using the map for anything official now, but would at least add a comment.

Reply:
Ah, that's right. I did mean to select the variable that indicates whether the DS was during the (actual) rainy season or not, the season_approx label. The original script for dry_spell_detection had that. Will follow up.

Reply:
The variable `during_rainy_season` uses rainy season onset and cessation dates to determine whether DSs were during the rainy season. Updated the filter ln 39 and ln 45 accordingly.